refactor(lsp): return Error on `initialize` request, when workspace uris are not valid file paths

Sysix opened the pull request:
Follow up from https://github.com/oxc-project/oxc/pull/17376#pullrequestreview-3613932883
Tell the client, the server will not be started when we can create "cwd" for the tools.

camc314 commented:
### Merge activity

* **Dec 27, 7:54 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 27, 7:55 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

Sysix committed:
refactor(lsp): return Error on `initialize` request, when workspace uris are not valid file paths (#17388)

Follow up from https://github.com/oxc-project/oxc/pull/17376#pullrequestreview-3613932883
Tell the client, the server will not be started when we can create "cwd" for the tools.
